static/app/views/issueList/overview.tsx

```diff
@@ -1215,6 +1213,27 @@ class IssueListOverview extends Component<Props, State> {
     };
   };
 
+  onStatsRequestPromise = (promise: Promise<Record<string, GroupStats>>) => {
+    promise
+      .then(data => {
+        if (!data || !data[0]) {
+          return;
+        }
+        this.trackTabViewed(this.state.groupIds, data[0] as unknown as Group[]);
+      })
+      .catch(err => {
+        this.setState({
+          error: parseApiError(err),
+        });
+      })
+      .finally(() => {
+        const currentTransaction = Sentry.getCurrentHub().getScope()?.getTransaction();
```

Comment:
Oof, this whole function makes things a bit complicated. I wonder if this could be extracted to a hook that checked for `data`, `isLoading`, and `isError` instead of needing the promise.

Reply:
Yeah, we need to expose the promise so that we can deal with the lifecycle tracking... Since this is only used in one place, we can always just inline it inside the provider, but it would likely need to be a callback because we want to propagate a state update to the parent component in case of an error.

Reply:
This was moved to useEffect